### PR TITLE
ROX-19956: adjust pipeline for roxctl@rhtap

### PR DESCRIPTION
**Pull request by msugakov**
My review on top of https://github.com/stackrox/stackrox/pull/7982.

It is best to review this PR by commits because I tried making them self-describing and isolated. Reviewing the entire thing may look too much.

### CODE REVIEW

File: .tekton/roxctl-push.yaml
Comment (tommartensen) [inline .tekton/roxctl-push.yaml:114]: ooops

**Review by tommartensen — COMMENTED** [review]
Nice work! Two points.

1. I would prefer to raise the issue with prefetch-input & hermetic builds (https://github.com/stackrox/stackrox/pull/7982/files/11cac29c276f0ab1b587292c626318ef0054710c#r1374239353, https://github.com/stackrox/stackrox/pull/8324/commits/c8bd7cd40d8b39dfad0166dce98b28ab5249076b) with RHTAP before changing the `when` condition to `prefetch-input`. I would assume there is a reason for their current implementation and why it did not work for you.

2. I agree that keeping 3 PipelineRuns is too few. Before bumping to 500, we should consult with the team about the implications.

**Comment by msugakov** [comment]
Regarding 2, PipelineRuns limit: asked in https://redhat-internal.slack.com/archives/C04PZ7H0VA8/p1698756980548709
I don't see this being an issue because Tekton provides a way to limit the value via `max-keep-run-upper-limit`, see <https://docs.openshift.com/pipelines/1.12/pac/install-config-pipelines-as-code.html#customizing-pipelines-as-code-configuration_install-config-pipelines-as-code>. RHTAP engineers could put a limit there to prevent us capturing all resources.

Regarding 1, `prefetch-dependencies`: I don't understand what the ask is about. Can you clarify, please?
`when:` is a Tekton syntax of simply gating a task execution with a conditional. https://tekton.dev/docs/pipelines/pipelines/#guard-task-execution-using-when-expressions
There's nothing specific to RHTAP in `when:` functioning itself. The expression we configure inside is specific though.
Previously it enabled prefetching on hermetic builds which is fair in most of cases: you must prefetch deps when building without networking (unless it's something extremely simple or all dependencies are vendored in a git repo).
However, the opposite behavior is not consequential: when you disable hermetic builds, prefetching also gets disabled. Why disabling prefetching, is there no value in it even without hermetic builds? I see many values in prefetching even without hermetic builds: builds speed up from more closely located dependencies source, protection from disappearing dependencies, vulnerability checks and controls (assuming Cachi2 has/will have that).
I simply changed the expression to enable/disable hermetic builds and prefetching independently of each other.

There is a possibility of a bug in some task implementation that `hermetic == false && prefetching == true` branch wasn't tested and does not work. I'm going to reproduce bugs and report but this takes us already in the area of ROX-20482 which I created. I'm not sure it's fair to request doing that and block the PR where an appropriate TODO is placed and the corresponding task visible in the backlog.

Also, turning the table, were you able to get prefetching correctly functioning when you were iterating on `roxctl` in <https://github.com/stackrox/stackrox/pull/8324>?

**Comment by msugakov** [comment]
>  Because the answer was not what we asked for, I am agreeing that we should go with 500 and see what breaks.

Cool. We seem to think the same about it. Let's break it early rather than such change becomes a problem.

> the `max-keep-runs` setting is propagated there

That's weird.

File: .tekton/roxctl-push.yaml
Comment (tommartensen) [inline .tekton/roxctl-push.yaml:236]: Let us tackle this end-to-end in the context of ROX-20482. 

```suggestion
      # TODO(ROX-20482): Enable dependency prefetching - this may need to be compared with prefetch-input instead
      - input: $(params.hermetic)
        operator: in
        values: [ "true" ]
```

File: .tekton/roxctl-pull-request.yaml
Comment (tommartensen) [inline .tekton/roxctl-pull-request.yaml:236]: Let us tackle this end-to-end in the context of ROX-20482. 

```suggestion
      # TODO(ROX-20482): Enable dependency prefetching - this may need to be compared with prefetch-input instead
      - input: $(params.hermetic)
        operator: in
        values: [ "true" ]
```

**Review by tommartensen — COMMENTED** [review]
Let's tackle the prefetch/hermetic discussion separately to unblock this PR.

File: .tekton/roxctl-push.yaml
Comment (msugakov) [reply .tekton/roxctl-push.yaml]: I left TODO note around `prefetch-input` parameter in this PR. I believe that should be sufficient.

File: .tekton/roxctl-pull-request.yaml
Comment (msugakov) [reply .tekton/roxctl-pull-request.yaml]: Same.

**Comment by msugakov** [comment]
@tommartensen I rebased and added two new commits at the end. You may want to check out the last one.

File: .tekton/roxctl-push.yaml
Comment (tommartensen) [reply .tekton/roxctl-push.yaml]: To be done in #8440 .

File: .tekton/roxctl-pull-request.yaml
Comment (tommartensen) [reply .tekton/roxctl-pull-request.yaml]: To be done in #8440 .

**Review by tommartensen — APPROVED** [review]
LGTM. Fine for me to merge to the onboarding branch.

**Pull request merged**
